code-challenges.test.js

```diff
@@ -14,50 +14,166 @@
 // Expected output: [-9, 7, 9, 199]
 
-var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+// var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+// Expected output: [-823, 7, 23]
+
+//copy a jest test template
+//make sure that the test fails
+// start a function 
+        // needs to take in array of junk data
+        // filiter just the number 
+        // filter just the odd number
+        // sort in order from least to greatest a-b
+//return array
+    // will propaly need a empty array that can hold odd numbers.
+//make sure fn passes
+
+// describe("array of mix junk", () => {
+//     junk = fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+//     oddNumbers = [-9, 7, 9, 199]
+//     it("returns an array of sorted odd numbers", () => {
+//       expect(junkYard(fullArr1)).toEqual([-9, 7, 9, 199])
+//       expect(junkYard(fullArr2)).toEqual([-823, 7, 23])
+//     })
+// })
+// const junkYard = (array) =>{
+//     numbersArray = array.filter(value =>{
+//         return (value % 2 !== 0 && value % 1 === 0 && value != true && value !== "7" && value !== "67") //&& value <= 0 && value >=0)
+//         })
+//     sortedArray = numbersArray.sort((a, b)=>
+//         a-b)
+//     return sortedArray
+// }
+
+//typeOf number
+
+// var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+// console.log(junkYard(fullArr1))
+// Expected output: [-9, 7, 9, 199]
+// var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+// console.log(junkYard(fullArr2))
 // Expected output: [-823, 7, 23]
 
 
 // b) Create the function that makes the test pass.
 
 
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+
+
+
+
+
+
 
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
+// var numbersToAdd1 = [2, 4, 45, 9]
+// // Excpected output: [2, 6, 51, 60]
+// var numbersToAdd2 = [0, 7, -8, 12]
+// // Expected output: [0, 7, -1, 11]
+// var numbersToAdd3 = []
+// // Expected output: []
+
+
+
+//make a shell test
+//let it fail
+// create a function
+    //sort the numbers fist
+    //maby use the iteration method
+    // adding i + i+i
+
+
+// describe("a function that takes in an array", () => {
+//     it("an array with the sum of all numbers", () => {
+//       expect([2, 4, 45, 9]).toEqual([2, 6, 51, 60])
+//     //   expect([]).toEqual([])
+//     //   expect([]).toEqual([])
+//     })
+// })
+
+const addMachine = (array)=>{
+    newSortedNumbers = array.sort((a, b)=>
+         a-b)
+    addedNumbers = []
+    for (let i = 0; i < newSortedNumbers.length; i++) {
+
+        return  + [[1]]
+    }
+    return newSortedNumbers
+}
```

Comment:
Okay, I am having trouble following along with the logic here-- What are we trying to do?